### PR TITLE
fix test_single_replica_restore_failure

### PR DESCRIPTION
**Pull request by meldafrawi**
Longhorn/Longhorn#1432

sometimes, replica instanceManagerName field is empty, causing crash replica function to fail.
we need to wait for volume state to be healthy, before crashing replicas

Signed-off-by: Mohamed Eldafrawi <mohamed.eldafrawi@rancher.com>

### CODE REVIEW

File: manager/integration/tests/test_ha.py [hunk 1]
```diff
@@ -1242,6 +1243,8 @@ def test_single_replica_restore_failure(client, core_api, volume_name, pod_make)
     res_volume = client.by_id_volume(res_name)
     assert res_volume.ready is False
 
+    res_volume = wait_for_volume_healthy_no_frontend(client, res_name)
+
     crash_replica_processes(client, core_api, res_name,
```
Comment (yasker) [inline manager/integration/tests/test_ha.py:1248]: We better add an `assert` in `crash_replica_processes` and similar functions that the instance manager must exist before continue. It can be a separate commit.

Comment (meldafrawi) [reply manager/integration/tests/test_ha.py:1248]: fixed in a separate commit.

Comment (yasker) [reply manager/integration/tests/test_ha.py:1248]: hmm? which one?

**Comment by yasker** [comment]
LGTM. Can you add the fix explanation to the commit?

**Comment by meldafrawi** [comment]
commit message updated.
